builder/config.go

```diff
@@ -23,38 +23,133 @@ import (
+var UseNewConfig = false
+
 // MixConfig represents the config parameters found in the builder config file.
 type MixConfig struct {
 	Builder builderConf
 	Swupd   swupdConf
+	Server  serverConf
 	Mixer   mixerConf
 }
 
 type builderConf struct {
-	BundleDir      string `required:"true"`
-	Cert           string `required:"true"`
-	ServerStateDir string `required:"true"`
-	VersionPath    string `required:"true"`
-	DNFConf        string `required:"true"`
+	BundleDir      string `required:"true" toml:"BUNDLE_DIR"`
+	Cert           string `required:"true" toml:"CERT"`
+	ServerStateDir string `required:"true" toml:"SERVER_STATE_DIR"`
+	VersionPath    string `required:"true" toml:"VERSIONS_PATH"`
+	DNFConf        string `required:"true" toml:"YUM_CONF"`
 }
 
 type swupdConf struct {
-	Format string `required:"true"`
+	Bundle     string `required:"false" toml:"BUNDLE"`
+	ContentURL string `required:"false" toml:"CONTENTURL"`
+	Format     string `required:"true" toml:"FORMAT"`
+	VersionURL string `required:"false" toml:"VERSIONURL"`
+}
+
+type serverConf struct {
+	DebugInfoBanned string `toml:"DEBUG_INFO_BANNED"`
+	DebugInfoLib    string `toml:"DEBUG_INFO_LIB"`
+	DebugInfoSrc    string `toml:"DEBUG_INFO_SRC"`
 }
 
 type mixerConf struct {
-	LocalBundleDir string `required:"false"`
-	LocalRepoDir   string `required:"false"`
-	LocalRPMDir    string `required:"false"`
+	LocalBundleDir string `required:"false" toml:"LOCAL_BUNDLE_DIR"`
+	LocalRepoDir   string `required:"false" toml:"LOCAL_REPO_DIR"`
+	LocalRPMDir    string `required:"false" toml:"LOCAL_RPM_DIR"`
+}
+
+// LoadDefaults sets sane values for the config properties
+func (config *MixConfig) LoadDefaults() error {
+	if !UseNewConfig {
+		return nil
+	}
+
+	pwd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	// [Builder]
+	config.Builder.BundleDir = filepath.Join(pwd, "mix-bundles")
+	config.Builder.Cert = filepath.Join(pwd, "Swupd_Root.pem")
+	config.Builder.ServerStateDir = filepath.Join(pwd, "update")
+	config.Builder.VersionPath = pwd
+	config.Builder.DNFConf = filepath.Join(pwd, ".yum-mix.conf")
+
+	// [Swupd]
+	config.Swupd.Bundle = "os-core-update"
+	config.Swupd.ContentURL = "<URL where the content will be hosted>"
+	config.Swupd.Format = "1"
+	config.Swupd.VersionURL = "<URL where the version of the mix will be hosted>"
+
+	// [Server]
+	config.Server.DebugInfoBanned = "true"
+	config.Server.DebugInfoLib = "/usr/lib/debug"
+	config.Server.DebugInfoSrc = "/usr/src/debug"
+
+	// [Mixer]
+	config.Mixer.LocalBundleDir = filepath.Join(pwd, "local-bundles")
+	config.Mixer.LocalRPMDir = ""
+	config.Mixer.LocalRepoDir = ""
+
+	return nil
 }
 
 // CreateDefaultConfig creates a default builder.conf using the active
 // directory as base path for the variables values.
 func (config *MixConfig) CreateDefaultConfig(localrpms bool) error {
+	if !UseNewConfig {
+		return config.createLegacyConfig(localrpms)
+	}
+
+	if err := config.LoadDefaults(); err != nil {
```

Comment:
You've lost `localrpms` here, which means running `mixer init --local-rpms --new-config` still leaves those two variables as empty strings.

Reply:
nice catch.